x-pack/platform/plugins/shared/security/public/management/roles/edit_role/privileges/kibana/privilege_summary/privilege_summary_table.tsx

```diff
@@ -48,14 +48,27 @@ function getColumnKey(entry: RoleKibanaPrivilege) {
   return `privilege_entry_${entry.spaces.join('|')}`;
 }
 
-function showPrivilege(allSpacesSelected: boolean, primaryFeature?: PrimaryFeaturePrivilege) {
+function showPrivilege({
+  allSpacesSelected,
+  primaryFeature,
+  globalPrimaryFeature,
+}: {
+  allSpacesSelected: boolean;
+  primaryFeature?: PrimaryFeaturePrivilege;
+  globalPrimaryFeature?: PrimaryFeaturePrivilege;
+}) {
   if (
     primaryFeature?.name == null ||
     primaryFeature?.disabled ||
-    (primaryFeature.requireAllSpaces && !allSpacesSelected)
+    (primaryFeature?.requireAllSpaces && !allSpacesSelected)
   ) {
     return 'None';
   }
+
+  if (primaryFeature?.requireAllSpaces && allSpacesSelected) {
+    return globalPrimaryFeature?.name ?? 'None';
```

Comment:
__nit:__ it'd be helpful if you can add a comment here with the example when the primary feature requires all spaces, all spaces are selected, but we cannot rely on then name of the primary feature and should use the name of the global primary feature instead